sisyphus/job.py

```diff
@@ -133,7 +133,7 @@ def __call__(cls: Type[T], *args, **kwargs) -> T:
 
         # add stacktrace information, if set to None or -1 use full stack
         stack_depth = gs.JOB_ADD_STACKTRACE_WITH_DEPTH + 1 if gs.JOB_ADD_STACKTRACE_WITH_DEPTH >= 0 else None
-        if stack_depth > 1 or None:
+        if stack_depth is None or stack_depth > 1:
             # add +1 for the traceback command itself, and remove it later
             stacktrace = traceback.extract_stack(limit=stack_depth)[:-1]
```

Comment:
So I personally would go with changing the meaning of `JOB_ADD_STACKTRACE_WITH_DEPTH = None` to handle this as disabled.
```suggestion
        if gs.JOB_ADD_STACKTRACE_WITH_DEPTH is not None and gs.JOB_ADD_STACKTRACE_WITH_DEPTH != 0:
            stacktrace = traceback.extract_stack(limit=gs.JOB_ADD_STACKTRACE_WITH_DEPTH if gs.JOB_ADD_STACKTRACE_WITH_DEPTH > 0 else None)
```

Reply:
if so, then lets remember to also change the default value in https://github.com/rwth-i6/sisyphus/blob/e61ba9ffb54e0fb421cc28d7c37457971663ed29/sisyphus/global_settings.py#L355

Reply:
0 also still means disabled with the proposed code change. So both 0 and None means deactivated.

Reply:
I checked internally the workings of `traceback.extract_stack`. Your implementation seems correct :+1:

Reply:
No `extract_stack(limit=None)` means to get the full stack. I think `extract_stack(limit=-1)` is not valid.

Reply:
> 0 also still means disabled with the proposed code change. So both 0 and None means deactivated.

Right.. but should we then just drop None from the permissible values and simply it even more?

> extract_stack(limit=-1) is not valid.

Negative numbers will print from the back of the stack
https://docs.python.org/3.10/library/traceback.html#traceback.extract_stack

Which brings me to another approach: just pass on the value of gs.JOB_ADD_STACKTRACE_WITH_DEPTH if it is different from 0 and then any negative number is allowed.
This would redefine the meaning of -1, keep the meaning 0 and None, and allow for more flexibility.

Reply:
It just feels a bit weird/confusing to me that the meaning of 0 is somehow the opposite of the meaning of None. But I don't really have strong opinions here.

The question is also, what have people used so far? We should try to avoid breaking existing setups. But I guess noone has used None, as that was broken anyway. The same for -1. So I just have set this to a high number (100) to enable it with full stack. But none of the discussed options would break any of that, so I guess we can decide for whatever we want.

Reply:
@Icemole You still did not consider this here. But there was also no clear conclusion here yet. Maybe we can really just not allow `None` for now, and require `int`.

Reply:
Thanks for the heads-up.

The issue is that, sadly, `depth=None` doesn't have the same semantical meaning as `limit=None`, so I wouldn't use `None`. It anyway has the same meaning as `0`.

I would personally only allow positive numbers (and zero). I also find `None` and `-1` a bit confusing, especially given the difference between "depth" and "limit".

How about the following?

```
stack_limit = max(gs.JOB_ADD_STACKTRACE_WITH_DEPTH, 0) if gs.JOB_ADD_STACKTRACE_WITH_DEPTH else 0
stacktrace = traceback.extract_stack(limit=stack_limit)
```

Reply:
Before we discuss the implementation (which is really trivial), let us discuss and agree on some wanted behavior. I summarized some potential behavior for all the possible cases [below](https://github.com/rwth-i6/sisyphus/pull/258#issuecomment-2871609864). Does everyone agree to follow that?

Reply:
> you suggested `-1` to be used as full stack as well.

No, this is outdated. The post [below](https://github.com/rwth-i6/sisyphus/pull/258#issuecomment-2871609864) is newer than that. Let's continue the discussion there (I moved it there because you seemed to have missed it here...). I think I discuss all possible cases there. Or do I miss anything?